### ISSUE TITLE
DreamBooth: Fine Tuning Text-to-Image Diffusion Models for Subject-Driven Generation

### ISSUE THREAD

**Issue by hypknot74** [hypknot74/Paper-summaries#42]
### リンク
arxiv: https://arxiv.org/abs/2208.12242
project page: https://dreambooth.github.io/
github: https://github.com/XavierXiao/Dreambooth-Stable-Diffusion (非公式？)

### 著者/所属機関
Nataniel Ruiz, Yuanzhen Li, Varun Jampani, Yael Pritch, Michael Rubinstein, Kfir Aberman
Google Research, Boston University

### 投稿日付
Submitted on 25 Aug 2022

## 概要
ある被写体の画像を数枚用意して、その被写体に対応した固有の識別子をtext-to-imageモデルで学習することで、その被写体を様々な生成画像に出現させることができる。
被写体に対応するテキストの埋め込み表現を探索する[texual inversion](https://textual-inversion.github.io/)とは異なり、text-to-image 拡散モデルをfine-tuningする。
<img width="985" alt="image" src="https://user-images.githubusercontent.com/57225520/204539232-35ba5d49-629a-4db3-8f05-d06f041a7b82.png">
<img width="992" alt="image" src="https://user-images.githubusercontent.com/57225520/204539391-afa7222b-926d-4846-bd34-c723b933a944.png">

## 手法
全体像はこの図で理解できる。
<img width="987" alt="image" src="https://user-images.githubusercontent.com/57225520/204930079-632ac6d4-f003-4a34-ab69-3be36c890bcf.png">

### Representing the Subject with a Rare-token Identifier
"a [identifier] [class noun]"のように入力画像の被写体をラベル付けする。

class nounはcatやdogのような大きな枠組みのクラス。identifierは被写体固有の識別子。
class nounがないと学習が遅い上に性能低下につながる。
<img width="992" alt="image" src="https://user-images.githubusercontent.com/57225520/204540077-0c355599-aa66-48a0-8df1-69b80f977a0d.png">

意味のある単語をidentifierに使うのは良くないが、"xxxy5syt00"のようなランダムな文字を選ぶのも良くない。トークナイザーは各文字を別々にトークン化する可能性があるため、意味がある文字列としてトークン化されるかもしれない。
1~3文字の短い文字列をidentifierに使うのが良い。

### Class-specific Prior Preservation Loss
純粋な学習を行うとoverfittingやlanguage drift(語彙を忘却する現象)が生じる。
Prior Preservation Lossを用いて学習することで両問題を解決する。

overfitting:
数枚での学習を行うと物体のポーズなどを過学習してしまう。実験亭に忠実性を最大化するのはモデル全体をfine-tuningした場合であることがわかった。
<img width="992" alt="image" src="https://user-images.githubusercontent.com/57225520/204540204-75de02e7-0542-4457-aa7c-a295caf60d4f.png">

language drift:
prior-preservation lossなしでは同一クラスの事前知識を忘却してしまい、多様な画像を生成できなくなる。
<img width="992" alt="image" src="https://user-images.githubusercontent.com/57225520/204540420-0e3381cc-6962-4f73-955b-a440bf5dd3b7.png">

#### Prior-Preservation Loss
overfittingとlanguage driftの問題を解決するためにautogenous class-specific prior-preserving lossを提案。
学習済みモデルが獲得している事前知識を保つことが目的。

固定した学習済み拡散モデルでfine-tuningに使用する[class noun]と同様のクラスの画像を生成し、fine-tuningするモデルでも"a [class noun]"でその画像を生成できるように学習する。(2)式の2項目がそれ。
<img width="992" alt="image" src="https://user-images.githubusercontent.com/57225520/204540969-7b47355b-d2e4-40b3-a644-f2e2a5433e7e.png">

### Personalized Instance-Specific Super-Resolution
SRモデルも再学習したほうが良い。
ただし、既存研究で提案されたノイズオーグメンテーションはあまり良くないことがわかったので、ノイズオーグメンテーションの程度を低くしてfine-tuningした。

### 異なる背景で対象の被写体を生成
<img width="975" alt="image" src="https://user-images.githubusercontent.com/57225520/204541499-75d39e1c-ac22-4237-ae5e-a30031ea35e9.png">

### 異なる画風で対象の被写体を生成
<img width="990" alt="image" src="https://user-images.githubusercontent.com/57225520/204541660-6e96adc9-79d2-468e-bc67-062994c0884d.png">

### 表情の編集
<img width="990" alt="image" src="https://user-images.githubusercontent.com/57225520/204541764-fdc386a4-aada-47cf-91a6-607d02cf3f4b.png">

### 向きの指定
<img width="990" alt="image" src="https://user-images.githubusercontent.com/57225520/204541863-591734ba-d5dc-4b5b-ab49-35717386e374.png">

### アクセサリーの指定
<img width="990" alt="image" src="https://user-images.githubusercontent.com/57225520/204541945-4d6b9c49-2482-4f33-abb6-b1437d997710.png">

### 被写体の属性の編集
<img width="990" alt="image" src="https://user-images.githubusercontent.com/57225520/204542215-cd5d1126-42ea-4b04-83a9-53115bacc7b5.png">

### textual inversionとの比較
<img width="994" alt="image" src="https://user-images.githubusercontent.com/57225520/204542551-ecdde78a-3e0b-40e3-a1fc-378f26f52693.png">

### 失敗例
<img width="994" alt="image" src="https://user-images.githubusercontent.com/57225520/204542687-7b5db234-2aa9-45ba-893c-941631647df4.png">

## コメント
